Set default desired peer count to 12

Eligioo opened the pull request:
## What's in this pull request?

Set the default value for the network desired number of peers to 12.
In the devnet scenarios the value is set to `min_peers` to keep previous behavior for testing to avoid several dial attempts when a node is taken down.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

sisou commented on web-client/src/client/lib.rs:
```diff
@@ -157,6 +157,7 @@ impl Client {
         config.network.seeds = seed_nodes;
         config.network.only_secure_ws_connections = true;
         config.network_id = web_config.network_id;
+        config.network.desired_peer_count = Some(8);
```
Where does `8` come from? Just a random number?

Eligioo replied:
For the most part yes. I've tested with various numbers and noticed that at some point, you won't notices a difference anymore. Since the amount of data a light client needs is relatively low, around 8 is a nice number whereas 10+ doesn't seem necessary. Going lower than 5, I had encounters where the consensus never got reached and the client kept syncing but always blocks behind. Also for Nimiq 1.0, the client roughly connects with the same amount of peers.

sisou replied:
This answer is to my satisfaction :+1:

sisou commented:
Here is a question to @jsdanielh though: Why does `desired_peer_count` fall back to `min_peer_count` in the first place. Those two have nothing in common in meaning. I think that is the root cause for this failure, and also affects all other node types (I think it even already came up in the Telegram dev channel).

sisou commented:
The default should be 12 when not specified otherwise, so why is this number not used for the light-client?
https://github.com/nimiq/core-rs-albatross/pull/2038/files#diff-19108d6db09124d07b1f465ebe19a763861d5fc69d7b490aaa23c3e963f83b28R78

jsdanielh commented:
> The default should be 12 when not specified otherwise, so why is this number not used for the light-client? https://github.com/nimiq/core-rs-albatross/pull/2038/files#diff-19108d6db09124d07b1f465ebe19a763861d5fc69d7b490aaa23c3e963f83b28R78

This is a different default that is defined internally in the network. The final network configuration is defined by the `Client`

jsdanielh commented:
Per our discussion today, we're going to fix this differently.

jsdanielh committed:
lib: Set the default desired number of peers to 12

Set the default value for the network desired number of peers to 12.
In the devnet scenarios the value is set to `min_peers` to keep
previous behavior for testing to avoid several dial attempts when a
node is taken down.

Eligioo reviewed (commented):
@jsdanielh LGTM
